server/paymentProviders/stripe/creditcard.js

```diff
@@ -308,11 +313,124 @@ export const setupCreditCard = async (paymentMethod, { user, collective } = {})
   return paymentMethod;
 };
 
+// Charge has a new Stripe dispute
+const createDispute = async event => {
+  const stripeChargeId = event.data.object.charge;
+  const transactions = await models.Transaction.findAll({
+    where: { data: { charge: { id: stripeChargeId } } },
+    include: [
+      {
+        model: models.Order,
+        required: true,
+        include: [models.Subscription],
+      },
+      {
+        model: models.Collective,
+        required: true,
+        as: 'collective',
+      },
+    ],
+  });
+
+  await Promise.all(
+    transactions.map(async transaction => {
+      await transaction.update({ isDisputed: true });
+      const order = transaction.Order;
+      if (order.status !== OrderStatuses.DISPUTED) {
+        await order.update({ status: OrderStatuses.DISPUTED });
+
+        if (order.SubscriptionId) {
+          await order.Subscription.deactivate();
+        }
+      }
+    }),
+  );
+};
+
+// Charge has been closed on Stripe (with status of: won/lost/closed)
+const closeDispute = async event => {
+  const stripeChargeId = event.data.object.charge;
+  const transactions = await models.Transaction.findAll({
+    where: { data: { charge: { id: stripeChargeId } }, isDisputed: true },
+    include: [
+      {
+        model: models.Order,
+        include: [models.Subscription],
+        required: true,
+      },
+    ],
+  });
+  const creditTransaction = transactions.find(tx => tx.type === 'CREDIT');
+
+  if (transactions.length > 0) {
+    const disputeStatus = event.data.object.status;
+    const disputeTransaction = event.data.object.balance_transactions.find(
+      tx => tx.type === 'adjustment' && tx.reporting_category === 'dispute',
+    );
+
+    // A lost dispute means it was decided as fraudulent
+    if (disputeStatus === 'lost') {
+      // Create refund transaction for the fraudulent charge
+      await createRefundTransaction(creditTransaction, 0, {
+        ...creditTransaction.data,
+        dispute: event,
+        refundTransactionId: creditTransaction.id,
+      });
+
+      // Create transaction for dispute fee debiting the fiscal host
+      const feeDetails = disputeTransaction.fee_details.find(feeDetails => feeDetails.description === 'Dispute fee');
+      const currency = feeDetails.currency.toUpperCase();
+      const amount = feeDetails.amount;
+      const fiscalHost = await models.Collective.findByPk(creditTransaction.HostCollectiveId);
+      const hostCurrencyFxRate = await getFxRate(currency, fiscalHost.currency);
+      const hostCurrencyAmount = Math.round(toNegative(amount) * hostCurrencyFxRate);
+
+      await models.Transaction.create({
```

Comment:
@vaughanj10, lets double check if OCF and OSC want to disclose this type of transaction in their ledger.

Reply:
Checked in with folks on Slack. Going to shift the language on this to something like "Stripe Review Fee"